packages/editor/src/components/document-outline/index.js

```diff
@@ -148,6 +150,7 @@ export default compose(
 			title: getEditedPostAttribute( 'title' ),
 			blocks: getBlocks(),
 			isTitleSupported: get( postType, [ 'supports', 'title' ], false ),
+			isTextModeEnabled: select( 'core/edit-post' ).getEditorMode() === 'text',
```

Comment:
The problem here, is that we can't use the `edit-post` store in the `editor` module (cycle dependency). This might be fixed by a prop for this component passed by the parent component in the `edit-post` module

Reply:
Isn't the same case for the Block navigation dropdown then?

https://github.com/WordPress/gutenberg/blob/cb809b98654151416646c56f5de42c93ce656315/packages/editor/src/components/block-navigation/dropdown.js#L51-L56

I've just used the same approach, but I guess it should be changed there too?

Reply:
yep, definitely wrong there too

Reply:
Good catch @youknowriad. I missed that. We definitely should pass this value down as a prop from a parent component in `edit-post` package instead 👍